docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,324 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/cluster-autoscaling.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Cluster Autoscaling
+
+## Preface
+
+This document briefly describes the design and roadmap of cluster autoscaling in Kubernetes.
+
+## Overview
+
+The purpose of Cluster Autoscaler is to adjust the size of the Kubernetes cluster to match
+the current needs of the applications running inside of the cluster. In general, if cluster
+nodes are under heavy load or there are pods that cannot be scheduled due to insufficient
+free resources then more machines should be added to the cluster. In the similar way if the
+machines are used only a little and it is apparent that the same set of pods could be run on
+fewer nodes, then the size of the cluster should be reduced.
+
+## Kubernetes 1.1 - no Kubernetes-native cluster autoscaling
+
+In the current Kubernetes version we have a basic support for cluster autoscaling. It is based
+on Google Cloud Autoscaler (https://cloud.google.com/compute/docs/autoscaler/) and works only
+in Google Compute Engine. It looks at total cpu usage on the node (including system stuff) 
+and total cpu request of all running pods. It can also scale on memory but for the simplicity
+this document will focus only on the CPU usage aspect (other metrics work in the exactly same way).
+The user is expected to provide the target level of utilization and GCA will adjust the size to 
```

Comment:
s/the size/the cluster size/

Is the "target level of utilization" referring to "total cpu usage on the node" or "total cpu request of all running pods"? Please clarify the input the user provides to GCA.